### PR TITLE
Add infested units

### PR DESCRIPTION
**Pull request by itsjustbones**
Adds the following

Renames old infested merc siege tank and banshees to infested siege breakers and dusk wings
adds coop stukov's infested factory, starport, and their tech labs (barracks not included, since there isn't any use for it)
adds coop stukov's infested siege tank, diamondback, liberator, and banshee.  Also adds volatile infested, which are used by the siege tank.
Adds the following upgrades
"Progressive Automated Mitosis (Infested Siege Tank)" - 2 levels, 1st is ammo every 30 seconds, 2nd is every 10 seconds (2nd level based on stukov prestige)
"Acidic Enzymes (Infested Siege Tank)" - +15 damage to armored units
"Deep Tunnel (Infested Siege Tank)" - deep tunnel to visible creep
"Frightful Fleshwelder (Zerg)" - 30% Resource efficiency for infested mech units, based on stukov prestige.  I thought about splitting it up into 4 upgrades, but didn't really want to without feedback
"Saturated Cultures (Infested Diamondback)" - reduce snare cooldown from 45 to 30 seconds
"Caustic Mucus (Infested Diamondback)" - adds slowing and damaging trail behind diamondback
"Braced Exoskeleton (Infested Banshee)" - +100 life
"Rapid Hibernation (Infested Banshee)" - allows burrowing, which gives increased life and energy regen
"Cloud Dispersal (Infested Liberator)" - changes the attack to be melee and gives 85% dr when active
"Viral Contamination (Infested Liberator)" - doubles primary attack target damage

I didn't apply other upgrades to the mech units.  Based on some leftover code in the stukov units, I decided to add some of the baneling upgrades to the volatile infected.  these might be too powerful, but they are cool, people recommended using banelings anyway, which would have caused this anyway (even worse, actually)
rupture  - 50% greater aoe size
corrosive bile - double direct target damage
healing explosion - heals 40 to friendlies in aoe

I didn't add screenshots here, since it's a lot of stuff and I didn't really want to make them if it's not needed, but let me know if you want them.

### CODE REVIEW

**Comment by Ziktofel** [comment]
Bug: Missing Banshee cloak/decloak sound

**Comment by Ziktofel** [comment]
Snímek obrazovky pořízený 2024-09-04 16-45-29
No hotkey for Enable/Disable structure attack

**Comment by Ziktofel** [comment]
I'd consider Infertor's infested Terrans being able to be used as ammo for tanks

**Comment by Ziktofel** [comment]
Tank's Volatile Burst could enjoy a tooltip telling the player what it exactly does

**Comment by Ziktofel** [comment]
Cloud Dispersal decreases attack range. There should remain a way to attack in the standard range after getting the item

**Comment by Ziktofel** [comment]
Snímek obrazovky pořízený 2024-09-04 17-01-32
Root/uproot tooltips need to be updated - Tank can attack in both modes but can launch volatile infesteds while rooted

**Comment by itsjustbones** [comment]
Should have fixed or implemented everything mentioned.  not 100% I fixed the siege tank issue though, I'm not sure what that actor error overlay is, but I'd like to know for the future.

**Review by Ziktofel — CHANGES_REQUESTED** [review]

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ActorData.xml [hunk 1]
```diff
@@ -35220,4 +35229,1122 @@
...
+        <Model value="AP_Infested_Liberator_Cloud"/>
+    </CActorModel>
+    <CActorSound id="AP_StukovInfestedLiberator_CloudImpactSound" parent="SoundOneShot">
+        <Sound value="AP_StukovInfestedLiberator_CloudImpact"/>
+    </CActorSound>
+    <CActorMissile id="AP_SILiberatorBaseAttackMissile" parent="GenericAttackMissile" unitName="AP_SILiberatorWeapon">
+        <Model value="AP_Stukov_Infested_Liberator_Missile"/>
+    </CActorMissile>
+    <CActorAction id="AP_SILiberatorBaseAttack" parent="GenericAttack" effectImpact="AP_SILiberatorSwarmAttackAB" effectLaunch="AP_SILiberatorBaseLM">
+        <On Terms="ActorCreation" Send="Create StukovInfestedLiberator_AttackLaunchSound"/>
```
[the remark this answers is not in the document]
Comment (itsjustbones) [reply Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ActorData.xml:36326]: fixed

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ActorData.xml [hunk 2]
```diff
@@ -35220,4 +35229,1126 @@
...
+    <CActorSound id="AP_StukovInfestedLiberator_CloudImpactSound" parent="SoundOneShot">
+        <Sound value="AP_StukovInfestedLiberator_CloudImpact"/>
+    </CActorSound>
+    <CActorMissile id="AP_SILiberatorBaseAttackMissile" parent="GenericAttackMissile" unitName="AP_SILiberatorWeapon">
+        <Model value="AP_Stukov_Infested_Liberator_Missile"/>
+    </CActorMissile>
+    <CActorAction id="AP_SILiberatorBaseAttack" parent="GenericAttack" effectImpact="AP_SILiberatorSwarmAttackAB" effectLaunch="AP_SILiberatorBaseLM">
+        <On Terms="ActorCreation" Send="Create AP_StukovInfestedLiberator_AttackLaunchSound"/>
+        <LaunchAttachQuery Methods="AMFilterWeaponBottom"/>
+        <ImpactMap index="None" Model="AP_Stukov_Infested_Liberator_Missile_Impact" Sound="AP_StukovInfestedLiberator_AttackImpact"/>
```
Comment (Ziktofel) [inline Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ActorData.xml:36328]: Missing this sound:
```xml
    <CSound id="StukovInfestedLiberator_AttackImpact" parent="Combat">
        <EditorCategories value="Race:Terran"/>
        <AssetArray File="Assets\Sounds\StukovInfestedLiberator_AttackImpact_01.wav" FacialGroup=""/>
        <AssetArray File="Assets\Sounds\StukovInfestedLiberator_AttackImpact_02.wav" FacialGroup=""/>
        <AssetArray File="Assets\Sounds\StukovInfestedLiberator_AttackImpact_03.wav" FacialGroup=""/>
        <AssetArray File="Assets\Sounds\StukovInfestedLiberator_AttackImpact_04.wav" FacialGroup=""/>
        <DupeDestroyCount value="3"/>
        <DupeMaximumMethod value="Replace"/>
        <DupeMuteCount value="5"/>
        <Pitch value="1.000000,2.000000"/>
        <Volume value="-9.000000,-7.000000"/>
    </CSound>
```

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ValidatorData.xml [hunk 3]
```diff
@@ -5957,4 +5957,145 @@
...
+        <Value value="AP_SIInfestedTrooperBurrowed"/>
+    </CValidatorUnitType>
+    <CValidatorCombine id="AP_IsAP_InfestorTerranCombine">
+        <CombineArray value="AP_IsAP_InfestorTerran"/>
+        <CombineArray value="AP_IsAP_InfestorTerranBurrowed"/>
+    </CValidatorCombine>
+    <CValidatorUnitType id="AP_IsAP_InfestorTerran">
+        <Value value="AP_InfestorTerran"/>
+    </CValidatorUnitType>
+    <CValidatorUnitType id="AP_IsAP_InfestedTerranBurrowed">
```
Comment (Ziktofel) [inline Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ValidatorData.xml:6023]: Typo: `AP_IsAP_InfestorTerranBurrowed`
Snímek obrazovky pořízený 2024-09-09 22-13-02

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ActorData.xml [hunk 4]
```diff
@@ -35220,4 +35229,1126 @@
...
+        <HostSiteOps Ops="SOpAttachOriginStationary SOpShadow"/>
+        <Model value="AP_StukovInfestedSiegeTankDeepTunnelBurrowDustModel"/>
+        <On Terms="Abil.AP_StukovInfestedSiegeTankDeepTunnel.SourceCastStop" Send="Create"/>
+    </CActorModel>
+    <CActorSound id="AP_Spell_StukovInfestedSiegeTank_DeepTunnel_Burrow" parent="SoundOneShot"/>
+    <CActorSound id="AP_Spell_StukovInfestedSiegeTank_DeepTunnel_Unburrow" parent="SoundOneShot"/>
+    <CActorRange id="AP_StukovInfestedSiegeTankSiegedRange">
+        <Weapon value="AP_StukovInfestedSiegeTank"/>
+        <On Terms="SelectionLocalUpdate.AP_StukovInfestedSiegeTank.Start; ValidateUnit AP_StukovInfestedSiegeTankAmmoGT0" Send="Create"/>
+        <On Terms="Behavior.AP_StukovInfestedSiegeTankAmmo.On; ValidateUnitType IsSISiegeTankSieged" Send="Create"/>
```
Comment (Ziktofel) [inline Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ActorData.xml:35930]: Snímek obrazovky pořízený 2024-09-09 22-17-45
Missing `AP_` Prefix in Validator

**Pull request merged**